melodies_monet/driver.py

```diff
@@ -459,6 +470,14 @@ def read_control(self, control=None):
                 self.control_dict['analysis']['output_dir'])
         self.debug = self.control_dict['analysis']['debug']
 
+        # generate time intervals for time chunking
+        if 'time_interval' in self.control_dict['analysis'].keys():
+            time_stamps = pd.date_range(
```

Comment:
This isn't necessarily an issue, but start_time and end_time must be separated by an integer multiple of time_interval. If that isn't the case then whole time period will not be captured within self.time_intervals. 

As an example say start_time='2019-09-05-00:00:00' and end_time='2019-09-16-00:00:00' and time_interval='1D'. The final the value of self.time_intervals is [2019-09-15 00:00:00', '2019-09-16 00:00:00'], so the final 16 hours are missed. It may just be worth adding a note about this somewhere so that users are aware of that.

Reply:
@colin-harkins did you mean end_time='2019-09-16-**16**:00:00'?

Reply:
@zmoon Yes, I did mean end_time='2019-09-16-16:00:00'. Thanks for catching that.

Reply:
We could move the readers outside of the time interval loop, and

```
from melodies_monet import driver

an = driver.analysis()
an.control = 'control_time_chunk.yaml'
an.read_control()

an.open_models()
an.open_obs()

for time_interval in an.time_intervals:
    an.pair_data(time_interval=time_interval)

an.concat_pairs()

an.plotting()
an.stats()
```

Reply:
But perhaps keep time_interval as an argument to the open_models and open_obs for now as well, to keep the flexibility in how to structure the top level time loop?